cpg_infra/abstraction/gcp.py

```diff
@@ -238,14 +252,18 @@ def create_bucket(
             unique_bucket_name = (
                 f'{self.config.dataset_storage_prefix}{self.dataset}-{name}'
             )
+
+        _lifecycle_rules = list(lifecycle_rules or [])
```

Comment:
If we allow `lifecycle_rules` being `None`, should this be reflected in the type hint as `list | None`?

Do we need the cast to `list` here?

Side nit: `versioning` and `project` parameters have a space for the default parameter assignment -- not sure why the linter doesn't pick that up.

Reply:
The list instantiation was to duplicate the list to avoid adding it to an existing list - I've refactored this to just be clearer and added a comment.